src/main/java/de/tum/in/www1/artemis/web/rest/StudentQuestionAnswerResource.java

```diff
@@ -123,20 +118,13 @@ public StudentQuestionAnswerResource(StudentQuestionAnswerRepository studentQues
             throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "idnull");
         }
         courseRepository.findByIdElseThrow(courseId);
-        Optional<StudentQuestionAnswer> optionalStudentQuestionAnswer = studentQuestionAnswerRepository.findById(studentQuestionAnswer.getId());
-        if (optionalStudentQuestionAnswer.isEmpty()) {
-            return ResponseEntity.notFound().build();
-        }
-        if (!optionalStudentQuestionAnswer.get().getQuestion().getCourse().getId().equals(courseId)) {
-            return forbidden();
-        }
-        if (mayUpdateOrDeleteStudentQuestionAnswer(optionalStudentQuestionAnswer.get(), user)) {
-            StudentQuestionAnswer result = studentQuestionAnswerRepository.save(studentQuestionAnswer);
-            return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, studentQuestionAnswer.getId().toString())).body(result);
-        }
-        else {
+        StudentQuestionAnswer optionalStudentQuestionAnswer = studentQuestionAnswerRepository.findByIdElseThrow(studentQuestionAnswer.getId());
+        if (!optionalStudentQuestionAnswer.getQuestion().getCourse().getId().equals(courseId)) {
             return forbidden();
         }
+        mayUpdateOrDeleteStudentQuestionAnswerElseThrow(optionalStudentQuestionAnswer, user);
+        StudentQuestionAnswer result = studentQuestionAnswerRepository.save(studentQuestionAnswer);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, studentQuestionAnswer.getId().toString())).body(result);
```

Comment:
Here we have `studentQuestionAnswer` and `optionalStudentQuestionAnswer` which now is also a `StudentQuestionAnswer `. Can you use better names and clarify why both are needed?

Reply:
I will fix the naming, i already changed that once but this got lost in a merge...
I need the optional (or soon existingStudentQuestionAnswer) to check if the course Id given as PathVariable matches the one in the RequestBody studentQuestionAnswer, to make sure the right answer of the right course gets changed